paasta_tools/kubernetes_tools.py

```diff
@@ -3680,28 +3722,72 @@ def update_crds(
     return success
 
 
-def get_kubernetes_secret_name(
-    service_name: str, secret_name: str, namespace: str = "paasta"
+def _get_secret_name(
+    namespace: str, secret_identifier: str, service_name: str, key_name: str
+) -> str:
+    return limit_size_with_hash(
+        "-".join(
+            [
+                namespace,
+                secret_identifier,
+                sanitise_kubernetes_name(service_name),
+                sanitise_kubernetes_name(key_name),
+            ]
+        ),
+        limit=253,
+    )
+
+
+def _get_secret_signature_name(
+    namespace: str, secret_identifier: str, service_name: str, key_name: str
 ) -> str:
-    service = sanitise_kubernetes_name(service_name)
-    sanitised_secret = sanitise_kubernetes_name(secret_name)
-    name = f"{namespace}-secret-{service}-{sanitised_secret}"
-    return name
+    """
+    Kubernetes secret names have character limit of 253 characters
+    """
+    return limit_size_with_hash(
+        "-".join(
+            [
+                namespace,
+                secret_identifier,
+                sanitise_kubernetes_name(service_name),
+                sanitise_kubernetes_name(key_name),
+                "signature",
+            ]
+        ),
+        limit=253,
+    )
 
 
-def get_kubernetes_secret(
+def get_paasta_secret_name(namespace: str, service_name: str, key_name: str) -> str:
+    return _get_secret_name(
+        namespace=namespace,
+        secret_identifier="secret",
+        service_name=service_name,
+        key_name=key_name,
+    )
+
+
+def get_paasta_secret_signature_name(
+    namespace: str, service_name: str, key_name: str
+) -> str:
+    return _get_secret_signature_name(
+        namespace=namespace,
+        secret_identifier="secret",
+        service_name=service_name,
+        key_name=key_name,
+    )
+
+
+def get_secret(
     kube_client: KubeClient,
-    service_name: str,
     secret_name: str,
+    key_name: str,
     namespace: str = "paasta",
     decode: bool = True,
 ) -> Union[str, bytes]:
-
-    k8s_secret_name = get_kubernetes_secret_name(service_name, secret_name, namespace)
-
     secret_data = kube_client.core.read_namespaced_secret(
-        name=k8s_secret_name, namespace=namespace
-    ).data[secret_name]
```

Comment:
isn't this a behavior change?

Reply:
oh, I think I see - `secret_name` is coming from `get_paasta_secret_name` which should be doing what `get_kubernetes_secret_name` was doing before for `k8s_secret_name` and `key_name` is now what `secret_name` was before

this just "looked" wrong from the diff :p

Reply:
I wonder if we should be a little more careful and add docstrings to anything dealing with secrets so that its clear how everything fits together without having to read the code?

Reply:
Added some docstring to `create_secret()` let me know if it's sufficient. Or should we create a wiki?

Reply:
I was thinking we could have docstrings on all of the functions we use for secret functionality so that it's clear how they all fit together + how they work (although, documenting in either our internal wiki or in the public paasta docs isn't a bad idea either as then we can add system diagrams and whatnot :))

that said, this isn't a blocker - but it'd definitely be nice to do (and in general, we should probably be writing more docstrings in paasta :p)